### ISSUE TITLE
Grokking

### ISSUE THREAD

**Issue by qt336** [dlphenomena/website#15]
---
description: A model initially memorizes training data but, after prolonged optimization, suddenly transitions to learning a more generalizable underlying structure.
title: Grokking
tags:
  - **You can choose from "Training; Neural network; LLM; ..."**
author: **Someone who submit the phenomenon to the website**
relatedWorks: related-works.json **A file to add the related-works to the submitted phenomenon, an example is located at ../content/phenomena/f-principle/ related-works.json**
---
## Introduction
Grokking is a training phenomenon observed in modern machine learning models—especially deep neural networks—where a model first appears to overfit (i.e., it memorizes the training data and performs poorly on unseen data), but then, after a long period of continued optimization without any change in training data or objective, it abruptly transitions to strong generalization.

A canonical example comes from work by Alethea Power and colleagues [1](#ref1), who studied small transformers trained on algorithmic tasks such as modular arithmetic. Early in training, the model achieves near-perfect training accuracy while test accuracy remains low, indicating memorization. However, after many additional training steps—sometimes orders of magnitude longer—the test accuracy suddenly rises to match the training accuracy. This sharp phase transition is what defines grokking.
## Detailed description in experiments and theoretical results.
Alethea Power train a 
## Key References
Please follow the reference style shown in the example located at ../content/phenomena/f-principle/f-principle.mdx.

<a id="ref1">[1]</a> Power, A.; Burda, Y.; Edwards, H.; Babuschkin, I.; Misra, V. _Grokking: Generalization Beyond Overfitting on Small Algorithmic Datasets_, arXiv preprint arXiv:2201.02177, 2022. [arXiv](https://arxiv.org/abs/2201.02177)

<a id="ref2">[2]</a> Xu, Zhi-Qin John; Zhang, Yaoyu; Luo, Tao; Xiao, Yanyang; Ma, Zheng. _[Frequency Principle: Fourier Analysis Sheds Light on Deep Neural Networks](https://global-sci.com/article/79739/frequency-principle-fourier-analysis-sheds-light-on-deep-neural-networks)_, Communications in Computational Physics, 2020, [arXiv](https://arxiv.org/abs/1901.06523)

<a id="ref3">[3]</a> Nasim Rahaman\#, Aristide Baratin\#, Devansh Arpit, Felix
Draxler, Min Lin, Fred A. Hamprecht, Yoshua Bengio, Aaron Courville. _[On the Spectral Bias of Neural Networks](https://proceedings.mlr.press/v97/rahaman19a.html)_, International Conference on Machine Learning, 2019, [arXiv](https://arxiv.org/abs/1806.08734)

<a id="ref4">[4]</a> Xu, Zhi-Qin John; Zhang, Yaoyu; Luo, Tao. _[Overview frequency principle/spectral bias in deep learning](https://link.springer.com/article/10.1007/s42967-024-00398-7)_,
Communications on Applied Mathematics and Computation, 2024, [arXiv](https://arxiv.org/abs/2201.07395)
